pkgs/native_assets_builder/lib/src/utils/run_process.dart

```diff
@@ -56,9 +42,7 @@ Future<RunProcessResult> runProcess({
     arguments,
     workingDirectory: workingDirectory?.toFilePath(),
     environment: environment,
-    includeParentEnvironment: includeParentEnvironment,
```

Comment:
We should strongly consider having
* `includeParentEnvironment: false` 
* have a minimal environment for the subprocess
* allow embedder to add extra environment variables

This avoids hooks to accidentally depend on things they shouldn't and would be a step towards a more hermetic/sandboxed build.

(Environment variables may expose also secrets - e.g. various access tokens / api keys / ... that we may not want to blindly pass to all the hook invocations)

Reply:
We'll at least need an allowlist for `PATH`, `TMP`, `TEMP`, `SYSTEMROOT`, and `ANDROID_HOME`.

See https://github.com/dart-lang/native/issues/32.

There might be more environment variables for other compilers and other use cases. Users will then either need to make a PR for this package and wait until it rolls or make PRs for the embedders.

> hermetic/sandboxed build.

With depending on `PATH` and the full file system for tools (clang, ndk, ...), I don't see how we can achieve hermetic builds.

But, I could change this PR easily to have an allowlist if we wanted.

Reply:
Even with an allowlist we'll need to invalidate the cache if any of the environment in the allow-list change. So let's merge this and do the allow-list in a follow up PR.